File: NXNavigationExtensionDemo/UIKit/Feature/Basic/Controllers/ViewController02_BackgroundImage.m
//
//  ViewController02_BackgroundImage.m
//  NXNavigationExtensionDemo
//
//  Created by Leo Lee on 2020/10/25.
//

#import <NXNavigationExtension/NXNavigationExtension.h>

#import "ViewController02_BackgroundImage.h"
#import "UIImage+NavigationBar.h"

@interface ViewController02_BackgroundImage ()

@end

@implementation ViewController02_BackgroundImage

- (void)viewDidLoad {
    [super viewDidLoad];
    
}

- (UIStatusBarStyle)preferredStatusBarStyle {
    return UIStatusBarStyleLightContent;
}

- (UIColor *)nx_barTintColor {
    return [UIColor whiteColor];
}

- (NSDictionary<NSAttributedStringKey,id> *)nx_titleTextAttributes {
    return @{NSForegroundColorAttributeName: [self nx_barTintColor]};
}

- (UIImage *)nx_navigationBarBackgroundImage {
    return UIImage.navigationBarBackgroundImage;
}

@end

File: Example/NXNavigationExtensionSwiftDemo/Basic/ViewController02_BackgroundImage.swift
//
//  ViewController02_BackgroundImage.swift
//  NXNavigationExtensionSwiftDemo
//
//  Created by Leo Lee on 2020/11/23.
//

import UIKit

class ViewController02_BackgroundImage: BaseTableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }
    
    override var nx_barTintColor: UIColor? {
        return .white
    }
    
    override var nx_titleTextAttributes: [NSAttributedString.Key : Any]? {
        return [NSAttributedString.Key.foregroundColor: UIColor.white]
    }
    
    override var nx_navigationBarBackgroundImage: UIImage? {
        return UIImage.navigationBarBackgorund
    }

}
